nibRealUI/Infobar/Blocks.lua

```diff
@@ -1161,67 +1161,75 @@ function Infobar:CreateBlocks()
 
                 -- Battle.net Friends
                 for i = 1, _G.BNGetNumFriends() do
-                    local bnetIDAccount, accountName, battleTag, _, characterName, bnetIDGameAccount, client, isOnline, _, isBnetAFK, isBnetDND, _, noteText = _G.BNGetFriendInfo(i)
-                    if isOnline then
-                        local _, _, _, _, _, _, _, class, _, zoneName, level, gameText, _, _, _, _, _, isGameAFK, isGameDND = _G.BNGetGameAccountInfo(bnetIDGameAccount)
-
-                        local name
-                        if accountName then
-                            name = accountName
-                            characterName = _G.BNet_GetValidatedCharacterName(characterName, battleTag, client)
-                        else
-                            name = _G.UNKNOWN
-                        end
-
-                        if characterName then
-                            if client == _G.BNET_CLIENT_WOW and _G.CanCooperateWithGameAccount(bnetIDGameAccount) then
-                                name = nameFormat:format(bnetFriendColor, name, _G.CUSTOM_CLASS_COLORS[ClassLookup[class]].colorStr, characterName)
+                    local accountInfo = _G.C_BattleNet.GetFriendAccountInfo(i)
+                    if accountInfo then
```

Comment:
I would collapse the if statement by also checking `isOnline` like this:
```suggestion
                    if accountInfo and accountInfo.gameAccountInfo.isOnline then
                    	local gameAccountInfo = accountInfo.gameAccountInfo
```

Then you can get rid of the second if and update later variables to use the new local table (mostly for readability). Other than that it looks good 👍